docs/ml-guide.md

```diff
@@ -801,6 +801,153 @@ jsc.stop();
 
 </div>
 
+## Examples: Summaries for LogisticRegression.
+
+Once [`LogisticRegression`](api/scala/index.html#org.apache.spark.ml.classification.LogisticRegression)
+is run on data, it is useful to extract statistics such as the
+loss per iteration which will provide an intuition on overfitting and metrics to understand
+how well the model has performed on training and test data.
+
+[`LogisticRegressionTrainingSummary`](api/scala/index.html#org.apache.spark.mllib.classification.LogisticRegressionTrainingSummary)
+provides an interface to access such relevant information. i.e the `objectiveHistory` and metrics
+to evaluate the performance on the training data directly with very less code to be rewritten by
+the user.
```

Comment:
We should document that logistic regression in ML currently [only supports two classes|https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L259] in the user guide (hence the casts to `BinaryLogisticRegression`), and that the traits are there for future extensibility purposes